engine/classes/Elgg/Application.php

```diff
@@ -0,0 +1,308 @@
+<?php
+
+namespace Elgg;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * @access private
+ *
+ * @since 2.0.0
+ */
+class Application {
+
+	/**
+	 * @var \Elgg\Config
+	 */
+	private $config;
+
+	/**
+	 * @var string
+	 */
+	private $engine_dir;
+
+	/**
+	 * @var string
+	 */
+	private $install_dir;
+
+	/**
+	 * @var string
+	 */
+	private $config_file;
+
+	public function __construct($config_file = '') {
```

Comment:
We might want to just allow injecting config totally here (no need to load config file), but this is still useful I think.